woocommerce/woocommerce-android
# [Manual Taxes M3] Integrating endpoint and handling auto tax rate setting

### Auto tax rate setting 

### Description

This PR allows setting up the auto tax rate setting, which will automatically populate the customer billing/shipping address when creating a new order. The goal of this feature is to allow automatic tax rate setting when a merchant is selling temporarily in a different location, e.g. at events like farmer’s markets, concerts, or craft fairs in different locations.

### Testing instructions

1. Verify tax rate selector works as expected - sets tax rate address data to the order
2. Verify auto tax rate setting works as expected - once turned on (see: bottom toggle in tax rate selector), the same address data is set in the next newly created orders automatically.

Known issue: Order update API bug - in case there is a product added to the order, deleting customer address data doesn't take effect.

### Images/gif

|Setting a tax rate|Modifying a tax rate|Auto rate setting|
|---|---|---|
|<video src=https://github.com/woocommerce/woocommerce-android/assets/4527432/c2e718b0-0a2c-4944-ab79-6958eb818713/>|<video src=https://github.com/woocommerce/woocommerce-android/assets/4527432/30007c01-667e-4e16-b923-5ec26110ec8e/>|<video src=https://github.com/woocommerce/woocommerce-android/assets/4527432/b010a97c-0de4-4763-b8bb-4f25f7f92653/>|
